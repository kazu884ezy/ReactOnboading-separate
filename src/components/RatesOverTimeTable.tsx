import React from 'react';
import {TableHead, TableBody, TableRow, TableCell} from "@material-ui/core";
import StripeTable from "./StripeTable";

interface RatesOverTimeTableProps {
    overtimeRates : string[][]
}

const RatesOverTimeTable = ({overtimeRates} : RatesOverTimeTableProps) => {
    const renderResults = (dataArray: string[][]) => {
        if(dataArray.length === 0) return;

        return dataArray.map(data => {
            let date = data[0];
            let currency = Object.keys(data[1]);
            let value = +Object.values(data[1]);
            return (
                <TableRow key={date}>
                    <TableCell>{date}</TableCell>
                    <TableCell>{currency} {value.toFixed(2)}</TableCell>
                </TableRow>
            );
        })
    }

    return (
        <StripeTable>
            <TableHead>
                <TableRow>
                    <TableCell>
                        Date
                    </TableCell>
                    <TableCell>
                        Value
                    </TableCell>
                </TableRow>
            </TableHead>
            <TableBody>
                {renderResults(overtimeRates)}
            </TableBody>
        </StripeTable>
    );
};

export default RatesOverTimeTable;
