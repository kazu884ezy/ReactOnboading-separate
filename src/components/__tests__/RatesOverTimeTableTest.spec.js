import React from 'react';
import {render, screen} from '@testing-library/react';
import RatesOverTimeTable from "../RatesOverTimeTable";

const testData = [
    ['2020-10-30', {AUD:1.1}],
    ['2020-10-31', {AUD:1.12}],
]

describe('RatesOverTimeTable test', () => {
    test('render correctly', () => {
        const { container } = render(<RatesOverTimeTable overtimeRates={testData} />);
        expect(container.firstChild).toMatchSnapshot();
        // screen.debug();
        expect(screen.getByText('AUD 1.10')).toBeInTheDocument();
        expect(screen.getByText('AUD 1.12')).toBeInTheDocument();
    })
})