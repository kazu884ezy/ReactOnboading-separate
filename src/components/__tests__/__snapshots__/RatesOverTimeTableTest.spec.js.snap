// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatesOverTimeTable test render correctly 1`] = `
<table
  class="MuiTable-root makeStyles-table-1"
>
  <thead
    class="MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Date
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Value
      </th>
    </tr>
  </thead>
  <tbody
    class="MuiTableBody-root"
  >
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        2020-10-30
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        AUD
         
        1.10
      </td>
    </tr>
    <tr
      class="MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        2020-10-31
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body"
      >
        AUD
         
        1.12
      </td>
    </tr>
  </tbody>
</table>
`;
